grommet/grommet
# Enhance Drop to account for align when assigning margin

#### What does this PR do?

#### Where should the reviewer start?
src/js/components/Drop/StyledDrop.js
#### What testing has been done on this PR?
Added the margin to custom example in storybook.

```
const customTheme = deepMerge(grommet, {
  global: {
    drop: {
      // margin: '10px',
      // intelligentMargin: true,
      background: { dark: 'neutral-2', light: 'neutral-2' },
      border: { radius: '10px' },
      zIndex: '13',
    },
  },
});



```
#### How should this be manually tested?
Add the `margin: '10px'` and `intelligentMargin: true`
to the custom theme in the custom storybook example. 
You can then see that depending on the alignment the margin is provided on the correct side.
#### Any background context you want to provide?
Margin was always given to all of the corners. This was something that we had to add to drop extend for the design system. 
The margin should be given depending on where the container is positioned to the target. I went through all of the scenarios where this would happen.
## Drop Align
Top: bottom
Bottom: top
Right: left
Left: right

Top: top Right: left
Top: top Left: right

Top: bottom Right: left
Top: bottom Right: right
Top: bottom Left: left 
Top: bottom Left: right

Bottom: top Right: left
Bottom: top Right: right
Bottom: top Left: left 
Bottom: top Left: right

Bottom: bottom Right: left
Bottom: bottom Left: right


#### What are the relevant issues?
Related to https://github.com/grommet/hpe-design-system/issues/1449
#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
already  updated
#### Should this PR be mentioned in the release notes?
yes I added this to the `global.drop.margin`
#### Is this change backwards compatible or is it a breaking change?
backwards compatible
